src/web/client/utilities/fileAndEntityUtil.ts

```diff
@@ -29,6 +29,11 @@ export function getFileEntityName(fileFsPath: string) {
         ?.entityName as string ?? WebExtensionContext.getVscodeWorkspaceState(fileFsPath)?.entityName as string;
 }
 
+export function getFileRootWebPageId(fileFsPath: string) {
+    const entityId = getFileEntityId(fileFsPath);
+    return WebExtensionContext.entityDataMap.getEntityMap.get(entityId)?.rootWebPageId as string ?? WebExtensionContext.getVscodeWorkspaceState(fileFsPath)?.rootWebPageId as string;
```

Comment:
Can we simplify this code for readability - and easy debugging- way too nested code

Reply:
I'm following the same convention followed in this file